server/web_ui/admin/src/components/admin_objectgraph.rs

```diff
@@ -0,0 +1,339 @@
+        }
+        sb.push('}');
+        init_graphviz(sb.as_str());
+
+        let node_refs = all::<ObjectType>()
+            .map(|object_type: ObjectType| { (object_type, NodeRef::default()) })
+            .collect::<Vec<_>>();
+
+        let on_checkbox_click = {
+            let scope = ctx.link().clone();
+            let node_refs = node_refs.clone();
+            move |_: Event| {
+                let mut filters = vec![];
+
+                for (obj_type, node_ref) in &node_refs {
+                    if let Some(input_el) = node_ref.cast::<HtmlInputElement>() {
+                        let checked = input_el.checked();
+
+                        if checked {
+                            filters.push(*obj_type);
+                        }
+                    }
+                }
+                scope.send_message(Msg::NewFilters { filters });
+            }
+        };
+
+        let view_graph_source = {
+            move |_: MouseEvent| {
+                open_blank(sb.as_str());
+            }
+        };
+
+        html! {
+            <>
+            <div class={CSS_PAGE_HEADER}>
+            <h2>{ "ObjectGraph view" }</h2>
+            </div>
+            if entries.is_empty() {
+                <div>
+                  <h5>{ "No graph objects for the applied filters." }</h5>
+                </div>
+            } else {
+                <div class="column">
+                    <div class="hstack gap-3">
+                    {
+                        node_refs.iter().map(|(ot, node_ref)| {
+                            let str = format!("{}", ot);
+                            let selected = filters.contains(&ot);
+
+                            html! {
+                                <>
+                                <div class="form-check">
+                                  <input class="form-check-input obj-graph-filter-cb" type="checkbox" ref={ node_ref } id={str.clone()} onchange={on_checkbox_click.clone()} checked={selected}/>
+                                  <label class="form-check-label" for={str.clone()}>{str.clone()}</label>
+                                </div>
+                                if *ot != ObjectType::last().unwrap() {
+                                    <div class="vr"></div>
+                                }
+                                </>
+                            }
+                        }).collect::<Html>()
+                    }
+                    </div>
+                    <button class="btn btn-primary mt-2" onclick={view_graph_source}>{ "View graph source" }</button>
+                </div>
+                <div id="graph-container" class="mt-3"></div>
+            }
+            </>
+        }
+    }
+
+    fn view_error(&self, _ctx: &Context<Self>, msg: &str, kopid: Option<&str>) -> Html {
+        error_page(msg, kopid)
+    }
+
+    async fn fetch_objects() -> Result<Msg, FetchError> {
+        let urls = vec!["/v1/group", "/v1/service_account", "/v1/person"];
```

Comment:
We could do this in oneshot to /v1/raw with a search request to get all of these at once. I think that also can request the entries are limited to certain attributes, which will reduce transfer size.

Reply:
Sounds good

Reply:
You should be able to use kanidm_proto SearchRequest + Filter to make something like:

```
Or(vec![
    Eq("class", "person"),
    Eq("class", "service_account"),
    Eq("class", "group")
])
```

Then post that to `/v1/raw/search`

Reply:
Limiting the response to certain attributes looks supported internally but not via search request,
see:
 - [handle_search](https://github.com/ToxicMushroom/kanidm/blob/b1e7cb13a54c8865154123bf3c7f90d51bf03cec/server/core/src/actors/v1_read.rs#L96)
 - [SearchEvent::from_message](https://github.com/ToxicMushroom/kanidm/blob/d5ed335b529d6eaff52c90a995342a32a7cb3dde/server/lib/src/event.rs#L100)

Reply:
But the oneshot request already works locally and simplified the request code a bit :)